pkg/filesystem/subvolume.go

```diff
@@ -87,7 +143,18 @@ func listCephFSSubvolumes(ctx context.Context, clientsets *k8sutil.Clientsets, o
 			continue
 		}
 		for _, svg := range subvolg {
-			svList, err := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "subvolume", "ls", fs.Name, svg.Name}, operatorNamespace, clusterNamespace, true)
+			cmd := "ceph"
+			args := []string{"fs", "subvolume", "ls", fs.Name, svg.Name, "--format", "json"}
+
+			if checkForExternalStorage() {
+				m, admin_id, admin_key := getExternalClusterDetails(ctx, clientsets, operatorNamespace, clusterNamespace)
+				cmd = "/bin/sh"
+				args = []string{
+					"-c",
+					fmt.Sprintf(`export CEPH_ARGS="-m=%s --id=%s --key=%s" && ceph fs subvolume ls %s %s --conf=/var/lib/rook/%s/%s.config --format json`, m, admin_id, admin_key, fs.Name, svg.Name, clusterNamespace, clusterNamespace),
```

Comment:
What is the reason to use `CEPH_ARGS`, and not pass the arguments to the `ceph` command directly? This looks more complex than what is needed, I think.

Is it not the same as this, with enhanced quoting of the `--key=` parameter?
```
ceph -m=%s --id=%s --key=%q fs subvolume ls %s %s --conf=/var/lib/rook/%s/%s.config --format json
```

Reply:
exporting to CEPH_ARGS is taken as a priority, I have tried earlier to pass them directly as you mentioned. But in this case its starts looking for file in which keyring is stored and hence fails to get them.

Reply:
@yati1998 as @nixpanic mentioned you will have the option to pass `--id= --keyring= --monitor=` explicitly if you want to connect to a different ceph cluster.